protocol/xmrmaker/errors.go

```diff
@@ -31,49 +30,49 @@ var (
 )
 
 type errBalanceTooLow struct {
-	unlockedBalance float64
-	providedAmount  float64
+	unlockedBalance *apd.Decimal
+	providedAmount  *apd.Decimal
 }
 
 func (e errBalanceTooLow) Error() string {
 	return fmt.Sprintf("balance of %s XMR is below provided %s XMR",
-		common.FmtFloat(e.unlockedBalance),
-		common.FmtFloat(e.providedAmount),
+		e.unlockedBalance.String(),
+		e.providedAmount.String(),
 	)
 }
 
 type errAmountProvidedTooLow struct {
-	providedAmount float64
-	minAmount      float64
+	providedAmount *apd.Decimal
+	minAmount      *apd.Decimal
 }
 
 func (e errAmountProvidedTooLow) Error() string {
 	return fmt.Sprintf("%s XMR provided by taker is under offer minimum of %s XMR",
-		common.FmtFloat(e.providedAmount),
-		common.FmtFloat(e.minAmount),
+		e.providedAmount.String(),
+		e.minAmount.String(),
 	)
 }
 
 type errAmountProvidedTooHigh struct {
-	providedAmount float64
-	maxAmount      float64
+	providedAmount *apd.Decimal
+	maxAmount      *apd.Decimal
 }
 
 func (e errAmountProvidedTooHigh) Error() string {
 	return fmt.Sprintf("%s XMR provided by taker is over offer maximum of %s XMR",
```

Comment:
```suggestion
	return fmt.Sprintf("%s ETH provided by taker is over offer maximum of %s XMR",
```
these are my bad lol